hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/TestSCMContainerPlacementRackAware.java

```diff
@@ -797,4 +797,44 @@ public void testNoFallbackWithUsedNodes(int datanodeCount) {
     Assertions.assertTrue(tryCount >= 1, "Not enough try count");
     Assertions.assertEquals(0, compromiseCount);
   }
+
+  @Test
+  public void chooseNodeWithUsedAndFavouredNodesMultipleRack()
+      throws SCMException {
+    int datanodeCount = 12;
+    setup(datanodeCount);
+    int nodeNum = 1;
+    List<DatanodeDetails> usedNodes = new ArrayList<>();
+    List<DatanodeDetails> favouredNodes = new ArrayList<>();
+
+    // 2 replica
+    usedNodes.add(datanodes.get(0));
+    usedNodes.add(datanodes.get(1));
+    // 1 favoured node
+    favouredNodes.add(datanodes.get(2));
+
+    List<DatanodeDetails> datanodeDetails = policy.chooseDatanodes(usedNodes,
+        null, favouredNodes, nodeNum, 0, 5);
+
+    Assertions.assertEquals(nodeNum, datanodeDetails.size());
+    // Favoured node should not be returned,
+    // as favoured node is in the same rack as both used nodes.
+    Assertions.assertFalse(favouredNodes.get(0).getUuid() ==
```

Comment:
Can it change to check that the returned node is not under the same rack as favouredNode?

Reply:
@ChenSammi, Thanks for the review. I have updated test to check different rack.